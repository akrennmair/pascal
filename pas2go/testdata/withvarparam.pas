program test;

type foox = record
			a : integer;
			b : real
		end;

var xx : foox;

procedure quux(var x : foox);
begin
	with x do
	begin
		a := 42;
		b := 23.5
	end
end;

begin
	quux(xx);
	writeln('xx.a = ', xx.a);
	writeln('xx.b = ', xx.b)
end.
