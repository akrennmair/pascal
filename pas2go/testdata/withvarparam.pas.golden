package main

import (
	"github.com/akrennmair/pascal/pas2go/system"
)

var _ = system.Write

// program test
func main() {
	type (
		foox struct {
			a int
			b float64
		}
	)

	var (
		xx foox
	)
	_ = xx

	var quux func(x *foox)
	quux = func(x *foox) {

		(*x).a = 42
		(*x).b = 23.5e0
		return
	}

	quux(&xx)
	system.Writeln("xx.a = ", xx.a)
	system.Writeln("xx.b = ", xx.b)
}
